### PR TITLE
Weekly Notes Dec 03, 2019

### PR DESCRIPTION
**Pull request by daviddias**

### CODE REVIEW

File: _meeting-notes/2019-12-03.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  - [Testing Infra Design](https://github.com/ipfs/testground/blob/master/docs/SPEC.md)
+  - [Testing Inventory](https://github.com/ipfs/testground/blob/master/docs/test-inventory.md)
+  - [Testing Improvement Tracker](https://docs.google.com/spreadsheets/d/1xyqyGUF-oe3x9ln88YonVeOMWWdknik74lVgL_3dBY8/edit#gid=0)
+
+:::
+
+## :mega: Update (Done, Shipped, Made Progress)
+> To be reviewed Async
+
+- ==@raulk==
```
Comment (daviddias) [inline _meeting-notes/2019-12-03.md]: @raulk, no leftover assignments from last week https://github.com/ipfs/testground/blob/master/_meeting-notes/2019-11-26.md#assignments 👍🏽

File: _meeting-notes/2019-12-03.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    - Little progress since I was on holiday last week, and now combating my backlog.
+    - Thought about multiversion test runs; planning to write a design blueprint.
+        - Gist: new concept "ensemble". An ensemble is a cohort of runs that run with the same ID but using different build outputs.
+        - Depends on tracking the dependency graph. Who can own this?
+    - Refactored the DHT plan. Currently a WIP PR; need to push it over the finish line this week.
+    - Arranging/negotiating travels for testground hack week.
+- ==@daviddias==
```
Comment (daviddias) [inline _meeting-notes/2019-12-03.md]: @daviddias, no leftover assignments from last week https://github.com/ipfs/testground/blob/master/_meeting-notes/2019-11-26.md#assignments 👍🏽

File: _meeting-notes/2019-12-03.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    - Created Onboarding Work Plan for Henrique
+    - Coordinated the handing of the Infra Responsibilities from Jim to Anton
+    - Created the MVP/Primary Goal issue https://github.com/ipfs/testground/issues/196
+    - Made the plans and handled the logistics for the next Team "Onboarding" Hack Week
+    - Created a PRD for the TestGround Dashboard
+    - Reviewed PRs
+    - Did not have any tasty Thanksgiving Holidays food (we need that holiday here! :D)
+- ==@stebalien==
```
Comment (daviddias) [inline _meeting-notes/2019-12-03.md]: @Stebalien, there are some leftover items from last week. Mind reviewing (highlighting if any progress or if blocked) and listing the ones you will be carrying to this week? These were:

- Sidecar: Support for Dropping Connections
- Sidecar: Support for Per-subnet constraints
- Create an issue describing the "multiple versions" design and a potential proposal including the 3 ones that got discussed last week.

Ref: https://github.com/ipfs/testground/blob/master/_meeting-notes/2019-11-26.md#assignments

File: _meeting-notes/2019-12-03.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    - Out Nov 27(Wed) - Nov 29(Fri) for Thanksgiving
+- ==@jimpick==
```
Comment (daviddias) [inline _meeting-notes/2019-12-03.md]: @jimpick, did you had a chance to do:

- Do the Makefile for Steven's Sidecar built

?? Will you carry it for this week? 

Ref: https://github.com/ipfs/testground/blob/master/_meeting-notes/2019-11-26.md#assignments

File: _meeting-notes/2019-12-03.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+        - Started on Test Plan 4 - "Nodes Connectivity (Transports, Hole Punching, Relay)" https://github.com/ipfs/testground/issues/96
+        - Figured out how to implement a NAT inside a (privileged) Docker container - does the sidecar model work for this?
+        - Wrote up some starter ideas ... "connectivity obstacle course"
+    - Updated smlbench2 - as requested, rebased and renamed to data-transfer-datasets-random
+        - https://github.com/ipfs/testground/pull/74
+        - Not working yet - networking fails, needs further investigation
+        - Can't be merged yet ... it makes nasty modifications to Dockerfile.template in pkg/build/golang - needs to be cleaner
+        - Doesn't map well to proposed tests written for data-transfer-datasets-random
+    - PR: Add status badges to plans https://github.com/ipfs/testground/pull/200
+- ==@nonsense==
```
Comment (daviddias) [inline _meeting-notes/2019-12-03.md]: @nonsense from https://github.com/ipfs/testground/blob/master/_meeting-notes/2019-11-26.md#assignments, did you get the chance to document how the daemon works?

File: _meeting-notes/2019-12-03.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+- [ ] Creation of a TestGround Dashboard that can aid the developers of go-ipfs & go-libp2p to grok the results outputted by every node in a TestGround run (owner: @daviddias)
+- [ ] Keep on iterating the DHT Test Plan (@owner: @daviddias)
+- [ ] Design blueprint for multi-version tests (aka test ensembles) (owner: @raulk)
+
+Full list can be at https://github.com/ipfs/testground/issues/196
+
+
+### Assignments
+
+- ==@raulk==
```
Comment (daviddias) [inline _meeting-notes/2019-12-03.md]: @raulk for your week assignments, consider the following asks from previous Weekly's

- Outline a plan that describes how the libp2p instrumentation will happen, how will it be built, what is the first experiment to be made and if possible, list who will own each part
- Revisit and refactor the SPEC to get it up to date
- Create a guilde line around Error Handling Patterns (ref: https://github.com/ipfs/testground/pull/78/files#r337779827)
- How should one write a test for when it is about ingesting multiple TBs of data (ref: https://github.com/ipfs/testground/pull/69#discussion_r337443200?)

File: _meeting-notes/2019-12-03.md [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+        - Collect architecture decisions that need to be captured and written down
+        - Create an Epic that tracks the Dependency Graph Tracking and link it to the go-ipfs 0.5.0 epic
+        - Link to HOW_WE_WORK from CONTRIBUTING.md
+- ==@stebalien==
+    - go-ipfs Hack Week
+    - For TestGround
+        - Last few changes on sidecar (the jitter PR + blocking inbound connection)
+        - Follow up on Jim's suggestion for Network Isolation (NAT) https://github.com/ipfs/testground/issues/96#issuecomment-559888146
+        - Write a Design Blueprint for Traffic Shapping (take inspiration from https://github.com/ipfs/testground/pull/203)
+- ==@jimpick==
```
Comment (daviddias) [inline _meeting-notes/2019-12-03.md]: @jimpick this week's plan is a bit shy. Please unpack it more with the work you are planning to do and if it happens to be the case that you don't know what to work on, follow the Kanban. Instructions at: https://github.com/ipfs/testground/blob/master/docs/HOW_WE_WORK.md

tl;dr; pick issues from the Ready Column that are not assigned to anyone https://app.zenhub.com/workspaces/test-ground-5db6a5bf7ca61c00014e2961/board

File: _meeting-notes/2019-12-03.md [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+    - Test Plan 4
+    - Test SideCar and demonstrate how to use it (docs or recorded demo)
+- ==@nonsense==
```
Comment (daviddias) [inline _meeting-notes/2019-12-03.md]: @nonsense similar to @jimpick's, can you unpack a bit more on what is your plan for the Week? I believe you have an issue coming up describing the "TestGround as a Service" vision, opening up for discussion (shared resource vs. a cluster for everyone)

File: _meeting-notes/2019-12-03.md [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+    - Merge the Infra PR from Jim
+    - Get the daemon PR merged
+- ==@hacdias==
+    - Dependency Graph Tracking
```
Comment (daviddias) [inline _meeting-notes/2019-12-03.md]:
```suggestion
    - Dependency Graph Tracking https://github.com/ipfs/testground/issues/147
```

File: _meeting-notes/2019-12-03.md [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+        - Write a Design Blueprint for Traffic Shapping (take inspiration from https://github.com/ipfs/testground/pull/203)
+- ==@jimpick==
+    - Test Plan 4
+    - Test SideCar and demonstrate how to use it (docs or recorded demo)
+- ==@nonsense==
+    - Merge the Infra PR from Jim
+    - Get the daemon PR merged
+- ==@hacdias==
+    - Dependency Graph Tracking https://github.com/ipfs/testground/issues/147
+    - Chew Large Datasets Plan
```
Comment (daviddias) [inline _meeting-notes/2019-12-03.md]: @hacdias can you unpack what you plan to do on this Test Plan for this week? There are multiple test cases that you could target, it is wiser to focus on a few at a time :)

**Comment by daviddias** [comment]
Hi folks! Just went through reviewing and iterating on these notes in a way that they can be useful to everyone.

In order to improve our weekly sprint process, I've a few asks moving forward:

- **You are responsible for both sharing what you got done during the week and also, what you didn't ship from the list of the previous assignments**. It's fine to say "I didn't do X", there can be a multitude of reasons, from got blocked, didn't get time, etc. Even if it is a "I wasn't able to", that's fine as well as we will be able to identify it and pair you with someone that can help you learn how to do it.
- **You are responsible for coming to the meeting with a set of tasks that you will be focusing int he following week** (3 is a good number as a minimum). In the end of the meeting, we will review if those still make sense.
- **As we move through the Questions, you are responsible to get the Action Items (AIs) identified and tag for you to your own assignments list**

Let me know if these are clear (an Ack or a 👍 will do) or if you have questions.

File: _meeting-notes/2019-12-03.md (hunk 4 above)
Comment (jimpick) [reply _meeting-notes/2019-12-03.md]: On Friday, I mentioned sidecar and you asked me not to work on it so I removed myself from the issue... https://github.com/ipfs/testground/issues/176

I can still do it - the Makefile seems simple enough, but I'm sure there are puzzles to figure out when it comes time to make sure that the right version is deployed before we run the runner (local and docker swarm).

File: _meeting-notes/2019-12-03.md (hunk 7 above)
Comment (jimpick) [reply _meeting-notes/2019-12-03.md]: I'm unhappy with the irregular granularity of the tasks in the Kanban board. The Ready column needs more triage. I see #96 in there ... but it's a gigantic epic ... clearly it needs to be broken into lots and lots of little tasks.

Comment (daviddias) [reply _meeting-notes/2019-12-03.md]: Indeed you are write, large tasks should be broken down.

I've converted all the other active "Impl Plan X" issues to epics, but issued this one. @jimpick given that you are owning it. Can you take the lead on unpacking the work that needs to be done or you want me to do it?

Comment (jimpick) [reply _meeting-notes/2019-12-03.md]: I can take the lead on it. Let's spend some 1:1 time to discuss it further - I've got some ideas.

Comment (daviddias) [reply _meeting-notes/2019-12-03.md]: Sounds good. Can you add it to your list of tasks for the week then?

File: _meeting-notes/2019-12-03.md (hunk 10 above)
Comment (hacdias) [reply _meeting-notes/2019-12-03.md]: I was thinking about upgrading the tests we already have by supporting multiple file sizes _and_ nested directories first, and then move to the others since I'd already have a structure for them. Sounds good? I can update the notes.

Comment (daviddias) [reply _meeting-notes/2019-12-03.md]: That sounds great! it would be great if you could get the one with sharding experiment enabled as well

Comment (hacdias) [reply _meeting-notes/2019-12-03.md]: Ah yes, I did enable the sharding thingy already... Forgot to mention that haha

Comment (hacdias) [reply _meeting-notes/2019-12-03.md]: I just need to add the file!

Comment (hacdias) [reply _meeting-notes/2019-12-03.md]: Updated!

File: _meeting-notes/2019-12-03.md [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+- **Host:** @daviddias
+- **Location:** https://protocol.zoom.us/j/181391682
+- **Date:** Dec 3, 2019 6:00 PM (UTC)
+- **Participants:**
+    - @daviddias
+    - @nonsense
+    - @raulk
+    - @jimpick
+    - @momack2
+    - @stebalien
```
Comment (nonsense) [inline _meeting-notes/2019-12-03.md]: Missing `@hacdias`.

File: _meeting-notes/2019-12-03.md (hunk 5 above)
Comment (nonsense) [reply _meeting-notes/2019-12-03.md]: I added godoc to the daemon package, but this is not very discoverable, unless you use Go daily.

Probably architecture descriptions belong to the docs/SPEC.md , but it is a bit outdated, and I didn't want to take the task to update it right now.

I still have it on my todo list, and once the architecture solidifies, we could update altogether the docs/SPEC.md and/or add more sections to the README.md

I hope this makes sense.

File: _meeting-notes/2019-12-03.md (hunk 8 above)
Comment (nonsense) [reply _meeting-notes/2019-12-03.md]: @daviddias updated the doc with a bit more granularity on the work to be done on `infra`. Not sure what you mean with the `TestGround as a Service` vision issue... I looked for this in the Github Issues, but I am not sure which one you're referring to...

Comment (daviddias) [reply _meeting-notes/2019-12-03.md]: Ah! fair point. It was indeed a note from our 1:1 back from Monday, perhaps it wasn't clear enough. The ask & recommendation made at the time was for you to feel empowered to own painting the picture and paving the path for what "TestGround as a Service" will be. In "my world view" that task would start as an issue and hence, me mentioning "an issue"

File: _meeting-notes/2019-12-03.md (hunk 10 above)
Comment (daviddias) [reply _meeting-notes/2019-12-03.md]: Note that sharding is for directories! :)

File: _meeting-notes/2019-12-03.md (hunk 4 above)
Comment (daviddias) [reply _meeting-notes/2019-12-03.md]: My apologies, I might have misunderstood. My intention was **avoid trying to duplicate or create overlapping work with what @Stebalien is doing/owning** and not advising to **drop a commitment made to a colleague without notice**.

I believe that @Stebalien is still counting on you to make that Makefile. It is best to check with him and update this mini thread and your weekly goals respectively.

File: _meeting-notes/2019-12-03.md (hunk 6 above)
Comment (raulk) [reply _meeting-notes/2019-12-03.md]: I can commit to 1 and 3 this week (assuming our weeks go Tue to Tue), possibly gathering some thoughts for 4 too, and initiating 1.

Comment (raulk) [reply _meeting-notes/2019-12-03.md]: Added those assignments in https://github.com/ipfs/testground/pull/203/commits/6a10cf415e624985a55790e1c1c0156d1875c28b

**Comment by daviddias** [comment]
Thank you for reviewing the async update, adding your assignments and ack'ing on the weekly responsibilities

image

Goooooooo TestGround Teaaaaaaam ❤️

**Comment by daviddias** [comment]
Created https://github.com/ipfs/testground/pull/211

File: _meeting-notes/2019-12-03.md [hunk 12]
```diff
@@ -139,6 +139,10 @@ Full list can be at https://github.com/ipfs/testground/issues/196
 
 - ==@raulk==
     - Review daemon PR.
+    - (HIGH PRIO) Outline candidate solutions and trade-offs for libp2p deep traceability.
+    - Initiate SPEC overhaul to align with actual implementation.
+    - Error handling guidelines.
+    - Start conceptualising multiple TB transfer test case design.
```
Comment (daviddias) [inline _meeting-notes/2019-12-03.md]: Ah! Read this one better now. It is not necessarily the TB transfer (@dirkmc is on top of that and @hacdias is on top of the adding part with #58). It is "what to do to have the multiple TB Harddrive so that we can run the test in AWS/locally, etc" @raulk

Comment (daviddias) [inline _meeting-notes/2019-12-03.md]: Ah! Read this one better now. It is not necessarily the TB transfer (@dirkmc is on top of that and @hacdias is on top of the adding part with #58). It is "what to do to have the multiple TB Harddrive so that we can run the test in AWS/locally, etc" @raulk
